lib/efi_loader/efi_bootmgr.c

```diff
@@ -355,15 +359,36 @@ static efi_status_t prepare_loaded_image(u16 *label, ulong addr, ulong size,
 	u64 pages;
 	efi_status_t ret;
 	struct udevice *ramdisk_blk;
+	struct blk_desc *desc;
+	struct part_driver *part_drv;
 
+	/* Create the ramdisk block device internally */
 	ramdisk_blk = mount_image(label, addr, size);
-	if (!ramdisk_blk)
-		return EFI_LOAD_ERROR;
+	if (!ramdisk_blk) {
+		log_err("Failed to create ramdisk block device\n");
+		return EFI_DEVICE_ERROR;
+	}
+
+	/* Get the block descriptor and detect partitions */
+	desc = dev_get_uclass_plat(ramdisk_blk);
```

Comment:
@apalos I know you said to do this:
```C
ret = prepare_loaded_image(....);
if (ret == EFI_SUCCESS) {
       struct blk_desc *desc = dev_get_uclass_plat(temp_bdev);
       part_drv = part_driver_lookup_type(desc);
}
```
But I wonder if moving everything to `prepare_loaded_image` as it's showing this change would be fine, as IMO the code looks cleaner.

Reply:
yes that looks ok